applications/client/src/views/Campaign/Explore/Panels/Comment/Comments.tsx

```diff
@@ -184,15 +184,16 @@ export const Comments = observer<CommentsProps>(({ sort }) => {
 	);
 
 	useEffect(() => {
+		// What does this do?
 		if (store.router.params.currentItem === 'comments_list' && store.router.params.currentItemId) {
-			const filteredData = commentsData?.presentationItems.find(
+			const currentPresentationItem = presentationItemsData?.presentationItems.find(
 				(item) => item.id === store.router.params.currentItemId
 			);
 			store.campaign?.setCommentsList({
-				commandGroupIds: Array.from(filteredData?.commandGroupIds || []),
+				commandGroupIds: Array.from(currentPresentationItem?.commandGroupIds || []),
 			});
 		}
-	}, [store.campaign.commentsList.commandGroupIds, store.router.params.currentItem, store.router.params.currentItemId]);
+	}, [presentationItemsData, store.router.params.currentItem, store.router.params.currentItemId]);
```

Comment:
@sharplessHQ, can you sort out what this useEffect is for? and also the required `presentationItemsData`? Could this be removed or is it important? Could we write a comment to explain the purpose?

Reply:
This is needed to render the page correctly under a specific comment type. Try refresh the page or sorting without this useEffect.

Reply:
- [ ] Sorting under a specific type won't work with this change.'

(As we talked yesterday, this dependency change would cause an issue and i used the circular way TEMPORARILY since i couldn't find another way to make it work at that moment. But i forgot what the issue was... Now i remember when checking your PR, it's the sorting under a specific type.)

Reply:
Got it, I will try to resolve it.

Reply:
I combined the 4 useQuery into 2 that provide different calls based on the requested state. I also removed the global `store.campaigns.commentList` because it is local to the `Comments.tsx` component.

BUT, this definitely needs review. @sharplessHQ and @GoldingAustin please look at the final state of [`Comments.tsx`](https://github.com/cisagov/RedEye/blob/f21d37cb6675b6e7adbdbca6d546172b7ee438b8/applications/client/src/views/Campaign/Explore/Panels/Comment/Comments.tsx) and see if it makes sense to combine these useQueries in this way. 

I've tested and everything seems to work fine. All ways returns the correct list, correctly sorted, and on initial load.

Reply:
It looks good and works right to me. I was planning to combine after i finished up the second set of queries, but then that refresh/sorting thing i mentioned above made me concerned combining might cause more issues. I guess my 2 separate sets of queries are the thing causing this issue...
Thanks for fixing this! :-)
My only comment is about the ProgressBar bug^